test/e2e/network/netpol/kubemanager.go

```diff
@@ -151,89 +145,25 @@ func (k *kubeManager) getPod(ctx context.Context, ns string, name string) (*v1.P
 	return kubePod, nil
 }
 
-// probeConnectivity execs into a pod and checks its connectivity to another pod.
+// probeConnectivity execs into a pod and checks its connectivity to another pod returning the true if the connectivity state is the expected.
 // Implements the Prober interface.
-func (k *kubeManager) probeConnectivity(args *probeConnectivityArgs) (bool, string, error) {
+func (k *kubeManager) probeConnectivity(args *probeConnectivityArgs) (bool, error) {
 	port := strconv.Itoa(args.toPort)
 	if args.addrTo == "" {
-		return false, "no IP provided", fmt.Errorf("empty addrTo field")
+		return false, fmt.Errorf("empty addrTo field")
 	}
-	framework.Logf("Starting probe from pod %v to %v", args.podFrom, args.addrTo)
-	var cmd []string
-	timeout := fmt.Sprintf("--timeout=%vs", args.timeoutSeconds)
-
-	switch args.protocol {
-	case v1.ProtocolSCTP:
-		cmd = []string{"/agnhost", "connect", net.JoinHostPort(args.addrTo, port), timeout, "--protocol=sctp"}
-	case v1.ProtocolTCP:
-		cmd = []string{"/agnhost", "connect", net.JoinHostPort(args.addrTo, port), timeout, "--protocol=tcp"}
-	case v1.ProtocolUDP:
-		cmd = []string{"/agnhost", "connect", net.JoinHostPort(args.addrTo, port), timeout, "--protocol=udp"}
-		if framework.NodeOSDistroIs("windows") {
-			framework.Logf("probing UDP for windows may result in cluster instability for certain windows nodes with low CPU/Memory, depending on CRI version")
-		}
-	default:
-		framework.Failf("protocol %s not supported", args.protocol)
+	negative := ""
+	if !args.expectConnectivity {
+		negative = "!"
 	}
-
-	commandDebugString := fmt.Sprintf("kubectl exec %s -c %s -n %s -- %s", args.podFrom, args.containerFrom, args.nsFrom, strings.Join(cmd, " "))
-
-	attempt := 0
+	framework.Logf("Starting probe from pod %v to %v", args.podFrom, args.addrTo)
+	cmd := []string{"timeout", "30", "/bin/sh", "-c", fmt.Sprintf(`until %s $( /agnhost connect --timeout=3s --protocol=%s %s ) ; do sleep 1; done`,
```

Comment:
@danwinship this is cheaper as only execs once in the pod, however is a bit more complex as we rely on the bash condition that may be harder to understand, but now the tests are much much faster